src/Microsoft.DotNet.ProjectModel/Compilation/AnalyzerAssembly.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel.Compilation
+{
+    public class AnalyzerReference
+    {
+        /// <summary>
+        /// The fully-qualified path to the analyzer assembly.
+        /// </summary>
+        public string AssemblyPath { get; }
+        
+        /// <summary>
+        /// The supported language of the analyzer assembly.
+        /// </summary>
+        public string AnalyzerLanguage { get; }
+        
+        /// <summary>
+        /// The required framework for hosting the analyzer assembly.
+        /// </summary>
+        public NuGetFramework RequiredFramework { get; }
```

Comment:
>  Since analyzers are design time, they don't care what the targets are.

That's false, you care about all of the same things that a regular target in the lock file expects. Actually, analyzers should really have their own section in the lock file all up with a new lock file.